R/estimate_proportion.R

```diff
@@ -32,14 +32,142 @@ prop_wilson <- function(rsp, conf_level, correct = FALSE) {
 }
 
 
+# help function for prop_strat_wilson
+strata_normal_quantile <- function(vars, weights, conf_level) {
+  summands <- weights^2 * vars
+  # Stratified quantile
+  sqrt(sum(summands)) / sum(sqrt(summands)) * qnorm((1 + conf_level) / 2)
+}
+
+#' @describeIn estimate_proportions Calculates the stratified Wilson confidence
+#'   interval for unequal proportions as described in [Yan and Su (2010)].
+#'
+#' @param strata (`factor`)\cr
+#'   with one level per stratum and same length as `rsp`.
+#' @param weights weights for each level of the strata. If missing, they are
+#'   estimated using the iterative algorithm proposed in [Yan and Su (2010)]
+#'   that minimizes the weighted squared length of the confidence interval.
+#' @param max_nit maximum number of iterations for the iterative procedure used
+#'   to find estimates of optimal weights.
+#' @param correct `logical`\cr
```

Comment:
```suggestion
#' @param correct (`flag`)\cr
```
flag = logical of length 1, cannot be missing etc.